app/models/effective/tagging.rb

```diff
@@ -0,0 +1,41 @@
+module Effective
+  class Tagging < ApplicationRecord
+    self.table_name = (EffectivePages.taggings_table_name || :taggings).to_s
+
+    belongs_to :tag, class_name: 'Effective::Tag'
+    belongs_to :taggable, polymorphic: true
+
+    validates :tag,      presence: true
+    validates :taggable, presence: true
+
+    scope :deep, -> { all }
+
+    effective_resource do
+      tag_id          :integer
+      taggable_id     :integer
+      taggable_type   :string
+
+      timestamps
+    end
+
+    validates :name, presence: true
```

Comment:
can we also validate uniqueness, so that the taggable can only belong to each tag once